api/include/opentelemetry/trace/tracer.h

```diff
@@ -162,6 +166,25 @@ class Tracer
     }
   }
 
+  /**
+   * Get the instrumentation library associated with the current tracer.
+   * @return the instrumentation library for current tracer.
+   */
+  InstrumentationLibrary *GetInstrumentationLibrary() const noexcept
+  {
+    return instrumentation_library_.get();
+  }
+
+  /**
+   * Set instrumentation library.
+   * @param instrumentation library to set on the current tracer.
+   */
+  void SetInstrumentationLibrary(
```

Comment:
As discussed, it should be possible to remove this method and instead pass `InstrumentationLibrary` instance in sdk::tracer constructor through sdk::tracerprovider::gettracer() . This would further need move creation of tracer instance in sdk::tracerprovider::gettracer() method.